src/encode.jl

```diff
@@ -0,0 +1,78 @@
+
+
+"""
+    gif_encode(filepath::AbstractString, img::AbstractArray; num::Int = 64)
+
+Encode the GIF colorant matrix to file. 
+
+#### Arguments
+- `filepath` : Name of the file to which image is written.
+- `img` : 3D GIF colorant matrix which has structure of height*width*numofimags and all the images are present as slices of the 3D matrix 
+- `colormapnum` : Specifies the number of colors to be used for the global colormap
+
+### Examples
+```jldoctest
+julia> using GIFImages, Downloads
+
+julia> path = "test/data/fire.gif"
+"test/data/fire.gif"
+
+julia> img = gif_decode(path)
+60×30×33 Array{RGB{N0f8},3} with eltype RGB{N0f8}
+
+julia> gif_encode("fire.gif", img)
+```
+"""
+function gif_encode(filepath::AbstractString, img::AbstractArray; colormapnum::Int = 256)
+    
+    if(eltype(img)!= RGB{N0f8})
+        throw(ErrorException("gif_encode requires img to be in RGB{N0f8} colorspace currently"))
+    end
+    
+    error = Cint(0)
+    gif_file = LibGif.EGifOpenFileName(filepath, 0, Ref(error))
```

Comment:
We should ensure this file is closed even if an error happens during the call. Or is the following pattern already done somewhere?

```julia
    gif_file = LibGif.EGifOpenFileName(filepath, 0, Ref(error))
    try
         ....
    finally
        if gif_file != C_NULL
            LibGif.EGifCloseFile(gif_file, 0)
        end
    end

Reply:
I have seen EGifCloseFile causing unusual errors but LibGif.EGifSpew(gif_file) works really well and close the file too, but its right of you to point that we should close the file in case a error happens n have a try and catch